Guide/Greedy/Interval Scheduling/README.md

```diff
@@ -0,0 +1,181 @@
+
+### Output Format
+- a list of the intervals existing in the optimal solution.
+
+## Algorithm
+### Overview
+For our interval scheduling algorithm we will re-state the problem in a more precise way:
+
+> Given a set _R_ = {i<sub>1</sub>...,i<sub>n</sub>} of requests where each request _i_ is associated with an interval [s(i), f(i)), find a best schedule, i.e. a subset _A_ of requests from _R_ such that:
+
+> 1. _A_ is _compatible_, i.e. _A_ contains no overlapping requests and 
+
+> 2. _A_ is _optimal_, i.e. there is no compatible subset of _R_ which has more requests than _A_.
+
+
+### Pseudo Code
+
+## Version 1
+
+```python
+
+    def interval_scheduling(set_of_requests):
+      1. While set_of_requests is not empty
+        a. select a request x from set_of_requests with earliest finish time
+        b. add x it to the solution set
+        c. remove all all sets incompatible with x from set_of_requests including x
+      2. return the solution set
+```
+
+## Version 2
+
+```python
+    def interval_scheduling(set_of_requests):
+        1. Sort set_of_requests in order of finishing time.
+        2. i = 0, n = len(set_of_requests)
+        3. f := − 1
+        4. while i <= n:
+            a. if set_of_requests[i]'s finish time is greater than or equal to f:
+                1. append the request to the solution set
+                2. f = the requests finish time
+            b. i = i + 1
+        5. return the solution set
+
+```
+
+### Analysis
+
+## Version 1
+
+Version 1 of the solution has a worst case time complexity of O(n<sup>2</sup>) and a best case of O(n). Since the list of requests is not sorted
+the algorithm must loop over the list of requests n times for the outer while loop, and n more times for finding the minimal finish time.
+
+## Example
+
+Requests = [(6, 9), (2, 3), (1, 4), (1, 3), (4, 5)]
+
+Solution = []
+
+Step 1:
+
+    a. first request with minimum finish time is (2, 3)
+    
+    b. Solution = [(2, 3)]
+    
+    c. Requests = [(6, 9), (4, 5)]
+
+Step 2:
+
+    a. request with minimum finish time is (4, 5)
+    
+    b. Solution = [(2, 3), (4, 5)]
+    
+    c. Requests = [(6, 9)]
+
+Step 3:
+
+    a. request with minimum finsih time is (6, 9)
+    
+    b. Solution = [(2, 3), (4, 5), (6, 9)]
+    
+    c. Request = []
+
+Solution set = [(2, 3), (4, 5), (6, 9)]
+
+## Version 2
+
+Version 2 of the solution has a worst case time complexity of O(n<sup>2</sup>) and an average time complexity of O(n log n) due to the sorting of the set of request before finding the optimal solution.
+The time complexity of the algorithm when the entire set of requests is sorted by finish time is simply n, and the time complexity of sorting
+the un-sorted set of requests is O(n log n), which means T(n) = O(n log n) + O(n) wherein the extra addition of n can be dropped.
+A faster sorting algorithm such as Radix sort which has a time complexity of O(kn) where k is the number of requests to be sorted, or
```

Comment:
Did you mean "n" instead of "k"? Also, from what I understand, Radix sort is only faster in certain circumstances. Maybe explain why it's faster in this case?